### PR TITLE
Add legacy note for spill to disk

### PR DESCRIPTION
**Pull request by mosabua**
## Description

This keeps coming up... might as well make it official. 

## Additional context and related issues

https://trinodb.slack.com/archives/CLEDR9V7G/p1722011311628259

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: docs/src/main/sphinx/admin/spill.md [hunk 1]
```diff
@@ -12,6 +12,12 @@ implemented on the application level to address specific needs of Trino.
 
 Properties related to spilling are described in {doc}`properties-spilling`.
 
+:::{warning} 
+The spill to disk feature and implementation are a legacy functionality of
+Trino. Consider using [](/admin/fault-tolerant-execution) with the `task` retry
```
Comment (mosabua) [inline docs/src/main/sphinx/admin/spill.md:17]: FTE is not really a replacement but solves the memory issue that spilling attempts to solve.

**Review by losipiuk — APPROVED** [review]

**Comment by wendigo** [comment]
@mosabua yeah, and we should probably start removing spilling logic as it's complex and broken anyway

**Review by wendigo — APPROVED** [review]

**Comment by mosabua** [comment]
> @mosabua yeah, and we should probably start removing spilling logic as it's complex and broken anyway

Sounds good. Wdyt @martint @dain @electrum - maybe we should create a roadmap issue and proceed?

**Review by electrum — APPROVED** [review]

**Comment by mosabua** [comment]
Here is the roadmap item - https://github.com/trinodb/trino/issues/22845

**Comment by losipiuk** [comment]
> @mosabua yeah, and we should probably start removing spilling logic as it's complex and broken anyway

complex - yeah. Broken????

We wanted to remove it from join some time ago - but athena is heavy user so we left forked operator.

**Comment by wendigo** [comment]
@losipiuk i recall some user reported issues that we left unanswered as we don't want to invest into spilling anymore

**Comment by sopel39** [comment]
> @losipiuk i recall some user reported issues that we left unanswered as we don't want to invest into spilling anymore

I don't think it's broken. It's also easier to deploy and has perf advantages as of now. I have second thoughts about removing it TBH.

**Pull request merged**